mkdocs/livereload/__init__.py

```diff
@@ -28,21 +28,37 @@
 
 _SCRIPT_TEMPLATE_STR = """
 var livereload = function(epoch, requestId) {
-    var req = new XMLHttpRequest();
-    req.onloadend = function() {
-        if (parseFloat(this.responseText) > epoch) {
-            location.reload();
-            return;
-        }
-        var launchNext = livereload.bind(this, epoch, requestId);
-        if (this.status === 200) {
-            launchNext();
-        } else {
-            setTimeout(launchNext, 3000);
-        }
-    };
-    req.open("GET", "/livereload/" + epoch + "/" + requestId);
-    req.send();
+    var req, timeout;
+
+    var poll = function() {
+        req = new XMLHttpRequest();
+        req.onloadend = function() {
+            if (parseFloat(this.responseText) > epoch) {
+                location.reload();
+            } else {
+                timeout = setTimeout(poll, this.status === 200 ? 0 : 3000);
+            }
+        };
+        req.open("GET", "/livereload/" + epoch + "/" + requestId);
+        req.send();
+    }
+
+    var stop = function() {
+        if (req)     req.abort()
+        if (timeout) clearTimeout(timeout)
+        req = timeout = undefined
+    }
+
+    window.addEventListener("load", function() {
+        if (document.visibilityState === "visible")
+            poll()
+    })
+    window.addEventListener("beforeunload", stop)
+    window.addEventListener("visibilitychange", function() {
+        var visible = document.visibilityState === "visible"
+        if (visible) poll()
+        else         stop()
+    })
```

Comment:
I would like a change to formatting
```suggestion
    var stop = function() {
        if (req) {
            req.abort();
        }
        if (timeout) {
            clearTimeout(timeout);
        }
        req = timeout = undefined;
    };

    window.addEventListener("load", function() {
        if (document.visibilityState === "visible") {
            poll();
        }
    });
    window.addEventListener("visibilitychange", function() {
        if (document.visibilityState === "visible") {
            poll();
        } else {
            stop();
        }
    });
    window.addEventListener("beforeunload", stop);
```

Reply:
That's much more verbose for my taste, but I'm fine with that if you like that better.